services/construction_service.go

```diff
@@ -156,6 +156,25 @@ func (s *ConstructionAPIService) ConstructionPreprocess(
 		preprocessOutputOptions.GasPrice = bigObj
 	}
 
+	// Override gas_limit
+	if v, ok := request.Metadata["gas_limit"]; ok {
+		stringObj, ok := v.(string)
+		if !ok {
+			return nil, wrapErr(
+				ErrInvalidGasLimit,
+				fmt.Errorf("%s is not a valid gas_limit string", v),
```

Comment:
Suggest instead:

fmt.Errorf("expected gas_limit value to be string, instead got: %T", v)

(you might drop the wrapping too as it probably doesn't add value if you do the above)